pyresample/bucket/__init__.py

```diff
@@ -234,11 +234,11 @@ def get_average(self, data, fill_value=np.nan, mask_all_nan=False):
         LOG.info("Get average value for each location")
 
         sums = self.get_sum(data, mask_all_nan=mask_all_nan)
-        counts = self.get_count()
+        counts = self.get_sum(np.logical_not(np.isnan(data)).astype(int),
+                              mask_all_nan=False)
```

Comment:
Should this type of logic be something that can be made a kwarg in `get_count`? Could this functionality be defaulted the way you want it, but be controlled by a kwarg?

Reply:
From our slack conversation: For buckets which we know will end up empty we have such a kwarg (``mask_all_nan``). But for the average-counts I wouldn't do that because the results are wrong. The only use case I can think of is speeding up the computation if the user knows that there are no NaNs in the data. Then ``get_average`` could directly use ``get_count``.